client/src/pages/Cart/components/CartItem/index.tsx

```diff
@@ -1,21 +1,56 @@
+import { ChangeEventHandler } from 'react';
+import { css, cx } from '@emotion/css';
+
 import { Button, Checkbox, Counter } from 'components';
 import { useCounter } from 'hooks';
+import { useCartActions, useCheckedCartIds } from 'store/cart';
 
+import { colors } from 'constants/colors';
 import { Cart } from 'types/cart';
 import { TrashSVG } from 'assets/svgs';
-import { css, cx } from '@emotion/css';
-import { colors } from 'constants/colors';
+
+import { useDeleteCarts } from '../../hooks';
 
 interface CartItemProps {
   cart: Cart;
+  refetchCarts: () => void;
 }
 
-function CartItem({ cart }: CartItemProps) {
-  const { count: defaultCount, product } = cart;
+function CartItem({ cart, refetchCarts }: CartItemProps) {
+  const { count: defaultCount, product, id } = cart;
   const { imageUrl, name, price } = product;
   const { count, minus, plus } = useCounter(defaultCount);
+  const { mutate: deleteCarts, isLoading } = useDeleteCarts({ onSuccess: refetchCarts });
+  const {
+    toggle: toggleIsChecked,
+    increase: increaseCartById,
+    decrease: decreaseCartById,
+  } = useCartActions();
```

Comment:
```suggestion
  const cartAction = useCartActions();
  
  cartAction.toggle()
```

취향의 영역인 것 같긴하지만, rename할때  rename이 같이 적용이 안돼서 위랑 같이 사용하는 것을 선호합니다.

Reply:
> 객체가 동작의 주체가 되어서 보다 명시적이라고 생각했어요

구조 분해 할당을 할 때마다 명시적이지 않다고 생각해서 rename을 하고 있었습니다. 그렇다고 useCartActions에서 반환해주는 함수의 네이밍을 위에서 제가 rename 한 것 처럼 바꾸기엔 그것도 뭔가 이미 useCartActions라는 훅의 이름에서 주체를 명시하고 있어서 그런지 불필요하게 두 번 명시하는 듯한 느낌을 받았거든요.

생각이 갇혀 있었던 것 같네요. 감사합니다.
이것도 추천해주신 방법이 마음에 들어서 적용해보도록 하겠습니다!

Reply:
[b4a140d](https://github.com/next-step/react-shopping-cart/pull/23/commits/b4a140de3a7f0f92d57fb35c26677995259d33b6)